contracts/SpokePool.sol

```diff
@@ -722,10 +723,8 @@ abstract contract SpokePool is SpokePoolInterface, Testable, Lockable, MultiCall
         bytes32 ethSignedMessageHash,
         bytes memory depositorSignature
     ) internal view virtual {
-        // Note: We purposefully do not support EIP-1271 signatures (meaning that multisigs and smart contract wallets
-        // like Argent are not supported) because of the possibility that a multisig that signed a message on the origin
-        // chain does not have a parallel on this destination chain.
-        require(depositor == ECDSA.recover(ethSignedMessageHash, depositorSignature), "invalid signature");
+        bool isValid = SignatureChecker.isValidSignatureNow(depositor, ethSignedMessageHash, depositorSignature);
```

Comment:
Maybe we should also add a comment about how `isValidSignatureNow` can change its behavior from block N to block M: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/cryptography/SignatureChecker.sol#L21

As long as the signature is valid at the time of calling `fillRelayWithUpdatedFee` then the signature is treated as valid.

Reply:
IMO, we should note that:
- EIP-1271 signatures are supported. This means that a signature valid now, may not be valid later.
- For an EIP-1271 signature to work, the depositor contract address must map to a deployed contract on the _destination_ chain that can validate the signature.
- Regular signatures from an EOA are also supported.

Reply:
Thanks for the input! I added them [here](https://github.com/across-protocol/contracts-v2/pull/221/commits/15cdaf60828edd6e2c56f3ef8c2fcdae31cfedd3)